reactor-core/src/main/java/reactor/core/publisher/FluxFlattenIterable.java

```diff
@@ -89,11 +90,11 @@ public void subscribe(CoreSubscriber<? super R> actual) {
 			}
 
 			Iterator<? extends R> it;
-
+			boolean knownToBeFinite;
 			try {
 				Iterable<? extends R> iter = mapper.apply(v);
-
 				it = iter.iterator();
+				knownToBeFinite = iter.spliterator().getExactSizeIfKnown() != FluxIterable.SPLITERATOR_UNSIZED;
```

Comment:
FYI I was looking at the sources of `java.util.Spliterator` and trying to find the `-1` constant there.

`getExactSizeIfKnown()` is defined as:
```java
    default long getExactSizeIfKnown() {
        return (characteristics() & SIZED) == 0 ? -1L : estimateSize();
    }
```

which means that we could also do `(iter.spliterator().characteristics() & Spliterator.SIZED) == 0` instead of calling `getExactSizeIfKnown()` which also matches what we're actually checking. WDYT?

Reply:
yeah that is still expressive enough and less computational-intensive